addons/libgprof/gprof.c

```diff
@@ -0,0 +1,49 @@
+/* KallistiOS ##version##
+
+    gprof.c
+    Copyright (C) 2024 Andy Barajas
+
+    This file contains functions that provide a workaround to get gprof
+    working on our current setup without editing startup.s and dealing with
+    trapa instructions. This implementation uses `-finstrument-functions` 
+    instead of `-pg` and requires linking against `-lgprof`.
+
+    Note: This implementation requires the following compilation and linking flags:
+
+    CFLAGS = -g -finstrument-functions
+    LDFLAGS = -lgprof
+*/
+
+#include <stdint.h>
+
+/* Start and End address for .text portion of program */
+#define BASE_ADDRESS 0x8c010000
```

Comment:
This should be tweaked slightly for NAOMI (0x8c020000) support. Both addresses are in `environ_dreamcast.sh` currently as part of `KOS_LDFLAGS`. Not sure if it makes more sense to modify the environ script and pass a define as a CFLAG or to just put both addresses here behind an ifdef. Alternatively I think you should be able to use `__executable_start` (similar to `_etext` below) which is provided by the linker script, might very slightly change the asm.

Reply:
Ah I missed that the only place `BASE_ADDRESS` gets used is in the setup function, probably just simplest to use `__executable_start` if that works. I initially assumed it was something being used for every sample and figured GCC could generate better code with a constant over a linker symbol.

Reply:
I'll use the __executable_start.  Thanks for pointing this out.  Forgot about Noami